src/read.ts

```diff
@@ -0,0 +1,122 @@
+import Mute from 'mute-stream'
+import { Completer, AsyncCompleter, createInterface, ReadLineOptions } from 'readline'
+
+export interface Options<T extends string | number = string> {
+    default?: T
+    input?: ReadLineOptions['input'] & {
+      isTTY?: boolean
+    }
+    output?: ReadLineOptions['output'] & {
+      isTTY?: boolean
+    }
+    prompt?: string
+    silent?: boolean
+    timeout?: number
+    edit?: boolean
+    terminal?: boolean
+    replace?: string,
+    completer?: Completer | AsyncCompleter,
+  }
+
+export async function read<T extends string | number = string> ({
+  default: def,
+  input = process.stdin,
+  output = process.stdout,
+  completer,
+  prompt = '',
+  silent,
+  timeout,
+  edit,
+  terminal,
+  replace,
+}: Options<T>): Promise<T | string> {
+  if (
+    typeof def !== 'undefined' &&
+    typeof def !== 'string' &&
+    typeof def !== 'number'
+  ) {
+    throw new Error('default value must be string or number')
+  }
+
+  let editDef = false
+  const defString = def?.toString()
+  prompt = prompt.trim() + ' '
+  terminal = !!(terminal || output.isTTY)
+
+  if (defString) {
+    if (silent) {
+      prompt += '(<default hidden>) '
+      /* c8 ignore start */
```

Comment:
There appear to be a LOT of new lines of coverage ignored.  Can we at least put a `TODO` above these to indicate that they're not intentional, and are due to the fact that they were never covered but ideally they should be?

Reply:
Good call, I'll add those.